### ISSUE TITLE
[csolution] A `pack:` specified with `path:` in YAML input takes precedence over the regularly installed pack version

### ISSUE THREAD

**Issue by jkrech** [Open-CMSIS-Pack/devtools#1098]
**Describe The Problem To Be Solved**
If the identical pack version is specified using the packID and by specifying a packID with a `path` resolving to the exact same version, the pack with path supersedes that regularly installed pack version.

```
  packs:
    - pack: MyVendor::MyPack@1.4.0
    - pack: MyVendor::MyPack
      path: ./MyPack 
```
Assuming that the latest version specified in ./MyPack/MyVendor.MyPack.pdsc is version 1.4.0, the pack from ./MyPack/... will be read and the pack from `$CMSIS_PACK_ROOT`/MyVendor/MyPack/1.4.0/... will be ignored.
Note: If the pack versions differ both packs will be read. Note that the packs can be specified in different YAML input files (*.csolution, *.cproject, *.clayer)

**Suggest A Solution**
A pack specified with `path` is always a pack with a fixed version, which needs to be read from the referenced `pdsc` file first. If multiple identical packIDs are specified a pack with `path` is supersedes a pack referenced from the CMSIS pack root directory or via the local_repository.pidx file.

**Comment by edriouk** [comment]
What do we need to do if multiple entries are specified 
```
packs:
    - pack: MyVendor::MyPack@1.5.0
    - pack: MyVendor::MyPack
      path: ./MyPack150 
    - pack: MyVendor::MyPack@1.4.0
    - pack: MyVendor::MyPack
      path: ./MyPack140 

```

**Comment by ReinhardKeil** [comment]
I believe this should result in an error when a `path:` is specified

**Comment by brondani** [comment]
>What do we need to do if multiple entries are specified

For the mentioned example the packs pointed by `./MyPack150` and `./MyPack140`, assuming their versions are respectively 1.5.0 and 1.4.0, should be loaded instead of the `MyPack@1.5.0` and `MyPack@1.4.0` from the pack repository. This is precisely what is requested in this GH issue. The third entry `./MyPack` should also be loaded if it has a third different version.
Since it's allowed to select multiple versions from the pack repository, for consistency we should not forbid multiple versions from local paths.

**Comment by edriouk** [comment]
Fixed in PR #1221
